osu.Game/Screens/OnlinePlay/DrawableRoomPlaylistItem.cs

```diff
@@ -470,6 +490,30 @@ protected override bool OnClick(ClickEvent e)
             return true;
         }
 
+        public MenuItem[] ContextMenuItems
+        {
+            get
+            {
+                List<MenuItem> items = new List<MenuItem>();
+
+                if (beatmap != null && collectionManager != null)
+                {
+                    if (downloadTracker.State.Value == DownloadState.LocallyAvailable)
+                    {
+                        var collectionItems = collectionManager.Collections.Select(c => new CollectionToggleMenuItem(c, beatmap)).Cast<OsuMenuItem>().ToList();
+                        if (manageCollectionsDialog != null)
+                            collectionItems.Add(new OsuMenuItem("Manage...", MenuItemType.Standard, manageCollectionsDialog.Show));
+
+                        items.Add(new OsuMenuItem("Collections") { Items = collectionItems });
+                    }
+                    else
+                        items.Add(new OsuMenuItem("Download to add to collection") { Action = { Disabled = true } });
```

Comment:
I really don't think this is required. It's just adding to the complexity of this whole thing without (imo) providing much use even if it did work.

Reply:
Instead of this, why not add the standard "Details..." to this context menu? I'd really hope that this becomes a standard context menu applied to any beatmap context going forward.

https://github.com/ppy/osu/blob/f8830c6850128456266c82de83273204f8b74ac0/osu.Game/Screens/Select/Carousel/DrawableCarouselBeatmapSet.cs#L223-L224